src/Yarhl.UnitTests/IO/DataStreamTests.cs

```diff
@@ -1387,6 +1387,111 @@ public void WriteToAfterDispose()
             Assert.Throws<ObjectDisposedException>(() => stream1.WriteTo("/ex"));
         }
 
+        [Test]
+        public void WriteSegmentToVariableLength()
+        {
+            DataStream stream1 = new DataStream();
+            stream1.WriteByte(0xCA);
+            stream1.WriteByte(0xFE);
+            stream1.WriteByte(0x00);
+            stream1.WriteByte(0xFF);
+            DataStream stream2 = new DataStream();
+            stream1.WriteSegmentTo(0, 2, stream2);
+            stream2.Position = 0;
+            Assert.AreEqual(0xCA, stream2.ReadByte());
+            Assert.AreEqual(0xFE, stream2.ReadByte());
+            Assert.IsTrue(stream2.Length == 2);
+            stream1.Dispose();
+            stream2.Dispose();
+        }
+
+        [Test]
+        public void WriteSegmentToNullFile()
+        {
+            DataStream stream1 = new DataStream();
+            stream1.WriteByte(0xCA);
+            stream1.WriteByte(0xFE);
+            stream1.WriteByte(0x00);
+            stream1.WriteByte(0xFF);
+            Assert.Throws<ArgumentNullException>(
+                () => stream1.WriteSegmentTo(0, 2, (string)null));
+            Assert.Throws<ArgumentNullException>(() => stream1.WriteSegmentTo(0, 2, string.Empty));
+            stream1.Dispose();
+        }
+
+        [Test]
+        public void WriteSegmentToFileAfterDispose()
+        {
+            DataStream stream1 = new DataStream();
+            stream1.WriteByte(0xCA);
+            stream1.WriteByte(0xFE);
+            stream1.WriteByte(0x00);
+            stream1.WriteByte(0xFF);
+            stream1.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => stream1.WriteSegmentTo(0, 2, "/ex"));
+        }
+
+        [Test]
+        public void WriteSegmentToFileCreatesParentFolder()
+        {
+            string tempFile = Path.Combine(
+                Path.GetTempPath(),
+                Path.GetRandomFileName(),
+                Path.GetRandomFileName());
+
+            DataStream stream = new DataStream();
+            stream.WriteByte(0xCA);
+            stream.WriteByte(0xFE);
```

Comment:
You need to write more data here so we can ensure we only write to the file a part of it.